docs/en/lfs.md

```diff
@@ -208,8 +147,12 @@ do
 end
 ```
 
-If you then create a file, say `LFS_dummy_strings.lua`, and put these `local preload` lines in it, and include this file in your `luac.cross -f`, then the cross compiler will generate a ROM string table that includes all strings referenced in this dummy module. You never need to call this module; just it's inclusion in the LFS build is enough to add the strings to the ROM table. Once in the ROM table, then you can use them subsequently in your application without incurring any RAM or LGC overhead.
-A useful starting point may be found in [lua_examples/lfs/dummy_strings.lua](../../lua_examples/lfs/dummy_strings.lua).
+You can do this at the interactive prompt or call it as a debug function during a running application in order to generate this string list, (but note that calling this still creates the overhead of an array in RAM, so you do need to have enough "head room" to do the call).
+
+You can then create a file, say `LFS_dummy_strings.lua`, and insert these `local preload` lines into it.  By including this file in your `luac.cross` compile, then the cross compiler will also include all strings referenced in this dummy module in the generated ROM string table.  Note that you don''t need to call this module; it's inclusion in the LFS build is enough to add the strings to the ROM table. Once in the ROM table, then you can use them subsequently in your application without incurring any RAM or LGC overhead.  A useful starting point may be found in [lua_examples/lfs/dummy_strings.lua](https://github.com/nodemcu/nodemcu-firmware/tree/dev/lua_examples/lfs/dummy_strings.lua); this saves about 4Kb of RAM by moving a lot of common compiler and Lua VM strings into ROM.
+
```

Comment:
It might be worth pointing out that the resulting LFS has a `LFS_dummy_strings` module in it that doesn't do anything.

Reply:
How is the magic that strings are in the rom table?
Every string that gets compiled by luac.cross is checked for duplicates and the ROM table and stored only once?
The node.compile also checks against the ROM table but cannot alter it (it's in ROM)
So it is only useful if I node.compile?
I would like to see some more explanation about that.

Reply:
>  it that doesn't do anything

Apart from freeing up about 4Kb RAM :wink:

@HHHartmann Gregor read on in the second half of the document.  Strings in Lua are [interned](https://en.wikipedia.org/wiki/String_interning).  The RTS will take any new string and resolve it first against the `ROstrt` and then the `RWstrt`.  If it already exists in one of those then this reference is used and so another copy is not created.  A new string is only created in the `RWstrt` in the case of a miss.  Also the Lua GC does mark and sweep on strings in the `RWstrt` so once dereferenced, they will be GCed again.

So if you compile or load an LC into RAM or even execute something like `dir..name..'.lua'`, this will still be checked against the `ROstrt` and created in the `RWstrt` if needed.  The strings in `dummy_strings.lua` are the ones that the Lua VM and the `file` and `net` modules create during startup.   I did think of adding the ones used in the compiler, as this would allow you to compiler larger files locally.

Reply:
OK. being used to GB of ram this seemed so crazy to me that I didn't even consider this.
Great job done!

Reply:
It's crazy what goes around comes around.  The first minis that I used were the PDP8 and PDP11 and the first micro the Intel 8080, and I did Motorola 6502 work and these sorts of tricks were standard, but discarded when VAXs and virtual memory came into vogue; they were all hauled out again with Win3.0 and the Intel 80286 protected mode, but abandoned again as 32-bit then 64-bit variants took over and VM became universal -- until IoTs turned up and they get rolled out again!

In essence we've gone from Lua code running in 48Kb RAM to running in 256Kb ROM + 48 Kb RAM, so you can now build apps 4× the size -- so long as you can work within the RAM constraint.